src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/infrastructure/clients/FeignErrorDecoder.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.iacasepaymentsapi.infrastructure.clients;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+@Component
+@Slf4j
+public class FeignErrorDecoder implements ErrorDecoder {
```

Comment:
We will probably want to review this later so it matches the kind of pattern we have elsewhere.  If we get a 4xx from the Fees we probably want to log it and then return a 5xx type error upstream.  But let's review and do in another PR.

Reply:
Sure